### ISSUE TITLE
Standard installation method is not working

### ISSUE THREAD

**Issue by ahilbers** [calliope-project/calliope#318]
## Problem description

I have a working distribution of `Calliope` on my machine, and it works great. However, I'm helping a colleague install it on his machine, and the install is not working. It seems the default is to import `Calliope 0.6.2`, and there are errors with the `enum` package. I've included the full traceback below.

## Steps to reproduce the problem

I use the standard installation from a command line:

```
conda create -c conda-forge -n calliope_alt calliope
```

which creates a new environment called `calliope_alt` with the package in it (I call it `calliope_alt` since my working distribution is an environment named `calliope`). However, when I open a python3 interpreter and try to import `calliope`, I get a set of errors:

```
(calliope_alt) $ python3
Python 3.6.10 | packaged by conda-forge | (default, Apr 24 2020, 16:27:41) 
[GCC Clang 9.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import calliope
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/calliope/__init__.py", line 2, in <module>
    from calliope.core import AttrDict, Model, read_netcdf, _logger, set_log_level
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/calliope/core/__init__.py", line 1, in <module>
    from calliope.core.model import Model, read_netcdf
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/calliope/core/model.py", line 28, in <module>
    from calliope.backend.run import run as run_backend
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/calliope/backend/run.py", line 14, in <module>
    import calliope.backend.pyomo.model as run_pyomo
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/calliope/backend/pyomo/model.py", line 16, in <module>
    import pyomo.core as po  # pylint: disable=import-error
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyomo/core/__init__.py", line 14, in <module>
    from pyomo.core.base import *
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyomo/core/base/__init__.py", line 11, in <module>
    import pyomo.core.kernel
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyomo/core/kernel/__init__.py", line 15, in <module>
    import pyomo.opt
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyomo/opt/__init__.py", line 14, in <module>
    from pyomo.opt.base import *
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyomo/opt/base/__init__.py", line 14, in <module>
    from pyomo.opt.base.convert import *
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyomo/opt/base/convert.py", line 16, in <module>
    from pyomo.opt.base.formats import guess_format
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyomo/opt/base/formats.py", line 16, in <module>
    from pyutilib.enum import Enum
  File "/Users/aph416/opt/anaconda3/envs/calliope_alt/lib/python3.6/site-packages/pyutilib/enum/__init__.py", line 18, in <module>
    package) that supersedes this library.""")
ImportError: pyutilib.enum has been removed.

Python 3 now has an enum implementation in the standard library (also
available for older Python versions as the third-party enum34 PyPI
package) that supersedes this library.
```

This seems to be related to [this error](https://github.com/Pyomo/pyomo/issues/1525), but I could not figure out a quick fix. On closer inspection, I see that my `Calliope` version is `0.6.2`. I've tried forcing `conda` to install a specific version using `calliope=0.6.5` or `calliope=0.6.4`, but this doesn't solve the problem.

## Calliope version

I'm trying to install 0.6.5 (I was originally trying for 0.6.4, but this seems to pop up for any version). However, `conda` seems to be installing 0.6.2.

**Comment by timtroendle** [comment]
I've ran into the same issue and a quick fix is to downgrade to pyutilib 5.8.0. euro-calliope has a [conda environment file](https://github.com/calliope-project/euro-calliope/blob/develop/envs/test.yaml) that gives you an environment avoiding this issue.

**Comment by ahilbers** [comment]
@timtroendle Thanks for the quick answer. If you'd like to close the issue, go ahead. Also happy to keep it open.

**Comment by timtroendle** [comment]
I think it should remain open as this needs a fix on Calliope's side as well.

**Comment by ahilbers** [comment]
In case anyone runs into this issue, running the install with the following command seems to work (credit to @jfallon1997, similar idea to @timtroendle 's above):

```
conda create -n calliope -c conda-forge calliope=0.6.5 python=3.8.1 numpy=1.16.5 pandas=0.25.3 pyomo=5.6.8 pyutilib=5.7.3 coincbc
```

**Comment by bbrannon4** [comment]
My colleagues and I have been having the same issue, and unfortunately this workaround isn't working, or at least bring up another error:

```
PackagesNotFoundError: The following packages are not available from current channels:

  - coincbc

Current channels:

  - https://conda.anaconda.org/conda-forge/win-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch
```

**Comment by brynpickering** [comment]
Just remove coincbc (referring to the CBC solver) from the list @bbrannon4, since it isn't available via conda forge for Windows

**Comment by ahilbers** [comment]
@bbrannon4 If you still wanted the optimizer `cbc` (in many settings it's quite a bit faster than the default `GLPK`), you can install it as explained [here](https://projects.coin-or.org/Cbc).

**Comment by brynpickering** [comment]
> @bbrannon4 In this case, if you still wanted the optimizer `cbc` (in many settings it's quite a bit faster than the default `GLPK`), you can install it as explained [here](https://projects.coin-or.org/Cbc).

Yep, CBC is [definitely a better option](https://calliope.readthedocs.io/en/stable/user/troubleshooting.html#influence-of-solver-choice-on-speed), we also refer to the manual installation of it on Windows in [our docs](https://calliope.readthedocs.io/en/stable/user/installation.html#solvers).

**Comment by sjpfenninger** [comment]
There seem to be two issues here:
* pyutilib, which pyomo depends upon, was updated with a backwards-incompatible change, leading to an error in pyomo. This should be fixed through #319.
* We don't recommend using Python 3.6 any more. There are all sorts of weird issues due to various package version constraints on conda when using older versions of Python and we can't account for those. In this case some dependency likely didn't exist on conda for Python 3.6 and macOS, causing conda to downgrade to an older version of Calliope. If an older version of Python is needed for some reason, it is advisable to install Calliope and dependencies through `pip` instead (which comes with its own issues, but seems less finicky than conda for combinations of older python versions and specific package versions being available).

**Comment by sjpfenninger** [comment]
With #319 merged this should be resolved.
